cluster-autoscaler/estimator/threshold_based_limiter.go

```diff
@@ -56,9 +60,24 @@ func (tbel *thresholdBasedEstimationLimiter) PermissionToAddNode() bool {
 // after either a node count- of time-based threshold is reached. This is meant to prevent cases
 // where binpacking of hundreds or thousands of nodes takes extremely long time rendering CA
 // incredibly slow or even completely crashing it.
-func NewThresholdBasedEstimationLimiter(maxNodes int, maxDuration time.Duration) EstimationLimiter {
+func NewThresholdBasedEstimationLimiter(limits []BinpackingLimit, maxDuration time.Duration) EstimationLimiter {
+	maxNodes := computeLimit(limits, 0)
```

Comment:
This does not work if any of the limits passed as the beginning is not constant, meaning that the limit it returns can change after some time. I think it would be better to store the limits here and calculate the maxNodes/maxDuration after each startEstimation.

Reply:
Thank you - I haven't accounted for limits that do computation on each invocation

Reply:
Fixed in 0d0196dbc6b41c145c34cfe91c3e3b769aabd9e9